protoc-gen-twirp/generator.go

```diff
@@ -923,10 +924,11 @@ func (t *twirp) generateServerMethod(service *descriptor.ServiceDescriptorProto,
 	methName := stringutils.CamelCase(method.GetName())
 	servStruct := serviceStruct(service)
 	t.P(`func (s *`, servStruct, `) serve`, methName, `(ctx `, t.pkgs["context"], `.Context, resp `, t.pkgs["http"], `.ResponseWriter, req *`, t.pkgs["http"], `.Request) {`)
-	t.P(`  switch req.Header.Get("Content-Type") {`)
-	t.P(`  case "application/json":`)
+	t.P(`  header := req.Header.Get("Content-Type")`)
+	t.P(`  switch {`)
+	t.P(`  case strings.HasPrefix(header, "application/json"):`)
```

Comment:
I don't think we want `HasPrefix` here - that would match bad stuff like `application/jsonp` incorrectly. Instead, I think we want to scan up to the first `;` character. Like

```
idx := strings.Index(header, ";")
if idx > 0 {
  header = header[:idx]
}
```
and then return to switching on equality matches.

Reply:
`jsonp` is a good point. I have occasionally see headers like `application/json+vnd.api` or something to indicate multiple parser options, but a brief skim leads me to believe that may be non-standard. Splitting on the `;` occurred to me too and I'm happy to do it this way. Expect an update this AM.

Reply:
The *most* right thing here might be to use https://golang.org/pkg/mime/ to parse the header. I'll leave you to decide whether that's overkill or worth it.

Reply:
Here's the [relevant mime function](https://golang.org/src/mime/mediatype.go?s=2954:3039#L102), and it basically does what you described, but also returns a bunch of parameters and junk we wont use. I think I'll just steal the top few lines:
```
i := strings.Index(v, ";")
  	if i == -1 {
  		i = len(v)
  	}
  	mediatype = strings.TrimSpace(strings.ToLower(v[0:i]))
```